### ISSUE TITLE
multi-APN with DHCP_broadcast mobilityd dies at boot

### ISSUE THREAD

**Issue by jpad-freedomfi** [magma/magma#3913]
### Your Environment

- **Version:** 1.3.2-1607014227-1462d960
- **Affected Component:** Access Gateway
- **Affected Subcomponent:** Mobilityd
- **Deployment Environment:** bare-metal (AGW)

### Describe the Issue

We are testing a dual APN, bridged (DHCP_broadcast) network with 1.3.2 tag version. See diagram for topology. 

When we boot the AGW, mobilityd dies and does not recover. Doing `sudo systemctl stop magma@*` does not work because mobilityd is so wedged. We have to do `sudo systemctl stop magma@mobilityd` and let it complete (~2-3 min) the do `sudo systemctl stop magma@*` then restart everything with `sudo systemctl start magma@magmad` after which things seem to come up normally and dual APN configuration is supported. 

**Network Diagram**
![AGW Bridge Dual SGi Test Net (4)](https://user-images.githubusercontent.com/61293007/101307794-88708700-3805-11eb-9e72-4e884a8139dc.png)

**To Reproduce**
1. Configure network for IP_Allocation = DHCP Broadcast
2. Configure network for multi-APN support
3. Configure Gateway for NAT disabled
4. Configure 2 APNs with 2 APN resources with VLAN tags
5. Allow config to propagate to AGW
6. Confirm you can connect any UE, dual APN or single
7. reboot AGW
8. after reboot, observer that no UE can complete Attach flow. 
9. try to stop all magma services with `sudo systemctl stop magma@*`
10. observer that all services are not stopped after prompt returns
11. force Mobilityd to stop `sudo systemctl stop magma@mobilityd` or kill PID
12. stop all remaining magma services `sudo systemctl stop magma@*`
13. restart all magma services with `sudo systemctl start magma@magmad`
14. try and attach a UE and see that now UE can attach

**Expected behavior**
Expected behavior is that all magma services come up to operational after reboot without manual intervention. 

**Network Config**
```
{
	"cellular": {
		"epc": {
			"default_rule_id": "default_rule_1",
			"gx_gy_relay_enabled": false,
			"hss_relay_enabled": false,
			"lte_auth_amf": "gAA=",
			"lte_auth_op": "EREREREREREREREREREREQ==",
			"mcc": "315",
			"mnc": "010",
			"mobility": {
				"enable_multi_apn_ip_allocation": true,
				"ip_allocation_mode": "DHCP_BROADCAST",
				"reserved_addresses": null
			},
			"network_services": [
				"policy_enforcement"
			],
			"tac": 1
		},
		"ran": {
			"bandwidth_mhz": 20,
			"tdd_config": {
				"earfcndl": 55340,
				"special_subframe_pattern": 7,
				"subframe_assignment": 1
			}
		}
	},
	"description": "test",
	"dns": {
		"enable_caching": false,
		"local_ttl": 0
	},
	"features": {
		"features": {
			"placeholder": "true"
		}
	},
	"id": "test_wiconnect",
	"name": "test"
}
```
**Gateway Config**

```
{
	"apn_resources": {
		"internet": {
			"apn_name": "internet",
			"gateway_ip": "192.168.9.1",
			"gateway_mac": "e0:63:da:22:47:21",
			"id": "nuc10_inet10",
			"vlan_id": 12
		},
		"management": {
			"apn_name": "management",
			"gateway_ip": "192.168.10.1",
			"gateway_mac": "e0:63:da:22:47:21",
			"id": "nuc10_mgmt10",
			"vlan_id": 13
		}
	},
	"cellular": {
		"dns": {
			"dhcp_server_enabled": true,
			"enable_caching": false,
			"local_ttl": 0
		},
		"epc": {
			"ip_block": "192.168.128.0/24",
			"nat_enabled": false,
			"sgi_management_iface_gw": "192.168.2.1",
			"sgi_management_iface_static_ip": "192.168.2.26/24",
			"sgi_management_iface_vlan": "2"
		},
		"non_eps_service": {
			"arfcn_2g": [
				0
			],
			"csfb_mcc": "315",
			"csfb_mnc": "010",
			"csfb_rat": 0,
			"lac": 1,
			"non_eps_service_control": 3
		},
		"ran": {
			"pci": 260,
			"transmit_enabled": true
		}
	},
	"connected_enodeb_serials": [
		"120200024019APP0105"
	],
	"description": "nuc10",
	"device": {
		"hardware_id": "413102cb-265e-467f-bd1f-b42575c5e4d2",
		"key": {
			"key": "MHYwEAYHKoZIzj0CAQYFK4EEACIDYgAEXT0fWU/wL/Y5wafObNbD7b5FTrPLiZQv7DFH1zQKcPXBKv+EIo5y9lxmlhFpAcDDYnSHCl4gCTA2yfFBsT0mq5sBfo6uUj40Mx4TN+RurOYxPxgJT++26K8TRg2+o6s9",
			"key_type": "SOFTWARE_ECDSA_SHA256"
		}
	},
	"id": "nuc10",
	"magmad": {
		"autoupgrade_enabled": true,
		"autoupgrade_poll_interval": 60,
		"checkin_interval": 60,
		"checkin_timeout": 30,
		"dynamic_services": [
			"eventd",
			"td-agent-bit",
			"smsd"
		],
		"logging": {
			"aggregation": {
				"target_files_by_tag": {
					"mme": "/var/log/mme.log"
				}
			},
			"log_level": "DEBUG"
		}
	},
	"name": "nuc10",
	"tier": "magma_1_3_2"
}
```
**Logst**
[1.3.2 mobilityd wedged at boot.txt](https://github.com/magma/magma/files/5650330/1.3.2.mobilityd.wedged.at.boot.txt)

**Comment by jpad-freedomfi** [comment]
@pshelar and @amarpad this effects ability to push 1.3.2 to WiConnect sites. Guessing you two are likely handy to tag.

**Comment by pshelar** [comment]
@jpad-freedomfi  mobilityD is not running due to missing dhcp0 port:
`
Dec 06 15:47:23 agw mobilityd[435]: OSError: b'dhcp0: No such device exists (SIOCGIFHWADDR: No such device)'
`
Can you check if the port is configured in OVS `ovs-vsctl show`.
also try restarting openvswitch service `sudo /etc/init.d/openvswitch-switch restart`

**Comment by berezovskyi-oleksandr** [comment]
@pshelar , I took a look on the boot sequence and in our setup openvswitch-switch starts later than mobilityD. This leads to the fact, that `dhcp0` has no possibility to be created once mobilityD starts and it is created later. 

Basically, I see two options here:
1) mobilityD self-stop in case of this problem wich will lead to eventual reboot.
2) Add `Wants=openvswitch-switch` and `After=openvswitch-switch.service` into `[Unit]` section of `magma@mobilityd.service`. I've verified that this approach works for me.
